tokio/src/sync/rwlock.rs

```diff
@@ -397,12 +397,19 @@ impl<T: ?Sized> RwLock<T> {
         }
     }
 
-    /// Locks this rwlock with shared read access, causing the current task
+    /// Locks this `RwLock` with shared read access, causing the current task
     /// to yield until the lock has been acquired.
     ///
-    /// The calling task will yield until there are no more writers which
-    /// hold the lock. There may be other readers currently inside the lock when
-    /// this method returns.
+    /// The calling task will yield until there are no writers which hold the
+    /// lock. There may be other readers inside the lock when the task resumes.
+    ///
+    /// Note that deadlock may occur if a read lock is held by the current
+    /// task, a write lock attempt is made, and then a further read lock
+    /// attempt is made. Under the priority policy of [`RwLock`], read locks
+    /// are not granted until prior write locks, to prevent starvation.
```

Comment:
I feel like this would read better if the order is reversed, i.e. first mention the fairness policy, and then warn that this can cause a deadlock if you try to acquire a read lock while holding a read lock.

Reply:
OK will do